### PR TITLE
Clear missing pgcon.pinned_by

### LINKED ISSUE #3170: Misleading error log when CLI disconnects in a transaction
- EdgeDB Version: 4aa682557093f7c4e44fefbcc1727333b59438dd
- OS Version: macOS

Steps to Reproduce:

1. Start a transaction in the CLI REPL
2. Terminate the CLI without closing the transaction

Server prints:

```
========================================= Exception occurred: cannot release connection <edb.server.pgcon.pgcon.PGConnection object at 0x11736cee0>: the connection was never acquired from the pool =========================================

1. RuntimeError: cannot release connection <edb.server.pgcon.pgcon.PGConnection object at 0x11736cee0>: the connection was never acquired from the pool

----------------------------------------------------------------------------------------------------------------- Traceback ------------------------------------------------------------------------------------------------------------------

    uvloop/cbhandles.pyx, line 69, in uvloop.loop.Handle._run
        > trimmed = Field(bool, default=False)
    uvloop/handles/basetransport.pyx, line 169, in uvloop.loop.UVBaseTransport._call_connection_lost
        > ExceptionContextList = checked.CheckedList[ExceptionContext]
    edb/server/pgcon/pgcon.pyx, line 2145, in edb.server.pgcon.pgcon.PGConnection.connection_lost
        > pinned_by.on_aborted_pgcon(self)
    edb/server/protocol/binary.pyx, line 331, in edb.server.protocol.binary.EdgeConnection.on_aborted_pgcon
        > self.server.release_pgcon(self.dbname, conn, discard=True)
    edb/server/server.py, line 521, in release_pgcon
        > self._pg_pool.release(dbname, conn, discard=discard)
    edb/server/connpool/pool.py, line 1156, in release
        > raise RuntimeError(

RuntimeError: cannot release connection <edb.server.pgcon.pgcon.PGConnection object at 0x11736cee0>: the connection was never acquired from the pool
ERROR 7468 2021-11-10T15:27:21.790 asyncio: Exception in callback UVTransport._call_connection_lost
handle: <Handle UVTransport._call_connection_lost>
INFO 7468 2021-11-10T15:27:21.796 edb.server: Connection discarded to backend database: edgedb
INFO 7743 2021-11-10T15:27:21.797 postgres: unexpected EOF on client connection with an open transaction
INFO 7468 2021-11-10T15:27:22.820 edb.server: Backend connections to database edgedb: 1 were established in at least the last 1.0 seconds.
```

### PR DESCRIPTION
**Pull request by fantix**
Fixes #3170

### CODE REVIEW

File: edb/server/protocol/binary.pyx [hunk 1]
```diff
@@ -366,6 +366,7 @@ cdef class EdgeConnection:
 
     cdef abort_pinned_pgcon(self):
         if self._pinned_pgcon is not None:
+            self._pinned_pgcon.pinned_by = None
             self._pinned_pgcon.abort()
```
Comment (1st1) [inline edb/server/protocol/binary.pyx:370]: Should this be moved directly into `PgCon.abort()`? Also, how did you stumble upon this?

Comment (fantix) [reply edb/server/protocol/binary.pyx:370]: > Should this be moved directly into `PgCon.abort()`?

Negative. In `PgCon.fallthrough_idle()`, we use `PgCon.abort()` to send `aborted_with_error` to EdgeDB clients.

> Also, how did you stumble upon this?

I was in a transaction in the CLI when quitting the CLI, and the server complained releasing the same pgcon twice. Then it's like `EdgeCon.abort_pinned_pgcon()` and `EdgeCon.on_aborted_pgcon() ` are both called.

Comment (1st1) [reply edb/server/protocol/binary.pyx:370]: > I was in a transaction in the CLI when quitting the CLI, and the server complained releasing the same pgcon twice. Then it's like EdgeCon.abort_pinned_pgcon() and EdgeCon.on_aborted_pgcon() are both called.

Can you replicate this in a test?

Comment (fantix) [reply edb/server/protocol/binary.pyx:370]: Sure, let me try.

**Review by 1st1 — APPROVED** [review]

**Pull request merged**